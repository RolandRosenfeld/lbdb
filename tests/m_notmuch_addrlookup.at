dnl Copyright (c) 2024-2025 Roland Rosenfeld <roland@spinnaker.de>
dnl
dnl This program is free software; you can redistribute it and/or modify
dnl it under the terms of the GNU General Public License as published by
dnl the Free Software Foundation; either version 2 of the License, or
dnl (at your option) any later version.
dnl
dnl This program is distributed in the hope that it will be useful,
dnl but WITHOUT ANY WARRANTY; without even the implied warranty of
dnl MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
dnl GNU General Public License for more details.
dnl
dnl You should have received a copy of the GNU General Public License along
dnl with this program; if not, see <https://www.gnu.org/licenses/>.

AT_BANNER([Test m_notmuch_addrlookup])

AT_SETUP(m_notmuch_addrlookup)
AT_SKIP_IF([! which notmuch-addrlookup])
AT_SKIP_IF([! which notmuch])
AT_CHECK([perl $abs_srcdir/gen-maildir $abs_srcdir/data
          echo "[[database]]" > notmuch.config
          echo "path=$(pwd)/maildir" >> notmuch.config
          echo "[[user]]" >> notmuch.config
          echo "primary_email=me@example.org" >> notmuch.config
          env NOTMUCH_CONFIG=notmuch.config notmuch new > /dev/null
          cat $abs_srcdir/data/m_notmuch_addrlookup.expected > expout
          env LBDB_OVERRIDE_METHODS=m_notmuch_addrlookup \
              NOTMUCH_CONFIG=notmuch.config \
          lbdbq example], [], [expout])
AT_CLEANUP
