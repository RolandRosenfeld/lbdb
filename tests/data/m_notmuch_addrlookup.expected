lbdbq: 16 matches
foo10@example.com		notmuch
foo11@example.com	Short11	notmuch
foo12@example.com	Short12	notmuch
foo13@example.com	foo13@example.com	notmuch
foo14@example.com	foo14@example.com	notmuch
foo15@example.com	Last15, First15	notmuch
foo1@example.com	First1 Family1	notmuch
foo2@example.com	foo2@example.com	notmuch
foo3@example.com	foo3@example.com	notmuch
foo4@example.com	First4 Family4	notmuch
foo5@example.com	foo5@example.com	notmuch
foo6@example.com	foo6@example.com	notmuch
foo7@example.com	foo7@example.com	notmuch
foo8@example.com	foo8@example.com	notmuch
foo9@example.com	foo9@example.com	notmuch
other@example.net	Other Other	notmuch
